packages/store/addon/-private/system/store/record-data-store-wrapper.ts

```diff
@@ -107,36 +110,46 @@ export default class RecordDataStoreWrapper implements IRecordDataStoreWrapper {
 
   inverseForRelationship(type: string, key: string): string | null {
     const modelClass = this._store.modelFor(type);
-    const definition = upgradeForInternal(this.relationshipsDefinitionFor(type)[key]);
+    const definition = this.relationshipsDefinitionFor(type)[key];
+    if (!definition) {
+      return null;
+    }
     if (CUSTOM_MODEL_CLASS) {
       if (definition.inverse !== undefined) {
         return definition.inverse;
       } else {
         //TODO add a test for this branch
-        if (!definition._inverseKey) {
-          return null;
+        if (metaIsRelationshipDefinition(definition)) {
+          return definition._inverseKey(this._store, modelClass);
         }
-        return definition._inverseKey(this._store, modelClass);
+        return null;
```

Comment:
Why null here now?

Reply:
it was null before too, we just inverted which thing was checked because it allows us to narrow what `definition` is. Now if there is a method for finding the inverse key we invoke it vs before if there was not a prop (that may or may not have been a method) we would return null.